internal/lib/lib.go

```diff
@@ -190,6 +190,11 @@ func GetNSXTTransportZone() string {
 func GetFqdns(vsName, key string, subDomains []string) ([]string, string) {
 	var fqdns []string
 	var fqdn string
+	// Only one domain is added for a Dedicated VS irrespective of
+	// the value given for the AutoFQDN.
+	if GetshardSize() == 0 {
```

Comment:
@swathinsankaran: From AviInfrasetting also we could get ShardSize. So we need to consider that also.

Reply:
I have added the code to consider `AviInfraSetting`. Thanks